app/controllers/application_controller.rb

```diff
@@ -1,10 +1,149 @@
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
-
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, 'secret'
+  end
+
+  get '/' do
+    if logged_in?
+      redirect to '/tweets'
+    else
+      erb :index
+    end
+  end
+  ####################### signup flow ##############
+  get '/signup' do
+    if logged_in?
+      redirect to '/tweets'
+    else
+      erb :'/users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if params.any? { |_k, v| v.empty? }
```

Comment:
i had the validates_presence_of :username in the user model all along. for some reason it kept getting  2 failed tests. I tried this and it worked. I think I would need to find out why validates pressance of doesn't work.